File: 141-linked-list-cycle/linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        s=set()
        crr=head
        while(crr!=None):
            if(crr in s):
                return True
            s.add(crr)
            crr=crr.next
        
        return False

File: 141-linked-list-cycle/linked-list-cycle.java

public class Solution {
    public boolean hasCycle(ListNode head) {
      
      HashSet<ListNode> seen=new HashSet<>();
      ListNode crr=head;
      while(crr!=null){
        if(seen.contains(crr)==true){
            return true;
        }
        seen.add(crr);
        crr=crr.next;
      }

      return false;

    }
}


/*
        ListNode slow=head,fast=head;
        while(fast!=null && fast.next!=null){
            slow=slow.next;
            fast=fast.next.next;

            if(slow==fast){
                return true;
            }
        }

        return false;
*/